packages/http-client-java/CONTRIBUTING.md

```diff
@@ -0,0 +1,51 @@
+# Contributing to TypeSpec Java Emitter Library
+
+## Prerequisite
+
+Install [Node.js](https://nodejs.org/) 20 or above. (Verify by running `node --version`)
+
+Install [Java](https://docs.microsoft.com/java/openjdk/download) 17 or above. (Verify by running `java --version`)
+
+Install [Maven](https://maven.apache.org/download.cgi). (Verify by running `mvn --version`)
+
+## Build and Pack
+
+["Setup.ps1" script](https://github.com/microsoft/typespec/blob/main/packages/http-client-java/Setup.ps1) builds TypeScript code and Java code, and packs them into "typespec-http-client-java-[version].tgz".
+
+One can use npm to install the `@typespec/http-client-java` tgz locally.
+
+```
+npm install [path-to-typespec-http-client-java-tgz]
+```
+
+## End-to-end Tests
+
+There is 2 end-to-end test modules:
+
+- `packages/http-client-java/generator/http-client-generator-clientcore-test` for testing of unbranded code
+- `packages/http-client-java/generator/http-client-generator-test` for testing of branded code
+
+Their folder structure is identical. In the following sub-sections, the script or command should be invoked in one of the two folders.
+
+### Set-up the environment
+
+"Setup.ps1" script builds the emitter, and sets up the test environment.
```

Comment:
Better to mention it internally calls the outer `Setup.ps1` to build the emitter and installs the tgz into test environment, to avoid confusion of the two scripts with same name..